### PR TITLE
feature- refactor with hooks

### PR DESCRIPTION
**Pull request by tufoinnkuo10**
In this project;
- The project was been refactored with Hooks
- Classes were replaced with Hooks function
- The useState was imported from react
- Event handler used
- Calculator runs fine in the browser as required

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @tufoinnkuo10 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,52 +1,62 @@
...
+  const [object, setObject] = useState({});
 
-  ifClicked = (e) => {
+  //  class Calculator extends React.Component {
+  //  constructor(props) {
+  //  super(props);
+  //  this.state = {};
+  //  }
+
+  // add event handler
```
Comment (mikenath223) [inline src/components/calculator.js]: - [x] Kindly remove this commented code.

File: src/components/calculator.js [hunk 2]
```diff
@@ -1,52 +1,62 @@
...
+    setObject(newObj);
   };
+  /*
+  ifClicked = (e) => {
+  const buttonName = e.target.textContent;
+  const newObj = calculate(this.state, buttonName);
+  this.setState(newObj);
+  };
+
+  render() { */
```
Comment (mikenath223) [inline src/components/calculator.js]: - [x] Also remove this commented code too.

File: src/components/calculator.js [hunk 3]
```diff
@@ -1,52 +1,62 @@
...
+    <div className="calc-container">
+      <div className="calc-screen">
+        {result}
+      </div>
+      <div className="buttons">
+        {buttonNames.map((button) => (<button key={button} className={getClassName(button)} onClick={ifClicked} type="button">{button}</button>))}
+      </div>
+    </div>
+  );
+};
```
Comment (mikenath223) [inline src/components/calculator.js:45]: - [ ] Please update the readme to match this [template](https://github.com/microverseinc/readme-template)

Comment (tufoinnkuo10) [reply src/components/calculator.js:45]: thanks for your time and review. I have made the required changes and updated the readme section.

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @tufoinnkuo10 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**